opm/simulators/utils/GridDataOutput_impl.hpp

```diff
@@ -0,0 +1,524 @@
+            auto xyz_local = vit.geometry().corner(0); // vertices only have one corner
+            x_inout[i] = static_cast<T>(xyz_local[0]);
+            y_inout[i] = static_cast<T>(xyz_local[1]);
+            z_inout[i] = static_cast<T>(0.0);
+            i++;
+        }
+    }
+    assert(i == nvertices_); // As we are templated on the
+                             // Dune::PartitionSet<partitions>, this cannot change
+    return i;
+}
+
+template <class GridView, unsigned int partitions>
+template <typename VectType>
+long SimMeshDataAccessor<GridView,partitions>::
+writeGridPoints(VectType& x_inout, VectType& y_inout, VectType& z_inout) const
+{
+    const std::size_t check_size_x = x_inout.size();
+    const std::size_t check_size_y = y_inout.size();
+    const std::size_t check_size_z = z_inout.size();
+
+    using VT = decltype(x_inout.data()[0]);
+
+    if ((check_size_x < static_cast<std::size_t>(nvertices_)) ||
+        (check_size_y < static_cast<std::size_t>(nvertices_)) ||
+        (check_size_z < static_cast<std::size_t>(nvertices_))) {
+        // assert(check_size >= nvertices_);
+        OPM_THROW(std::runtime_error,
+                  "Opm::GridDataOutput::writeGridPoints( VectType&  x_inout,  VectType&  "
+                  "y_inout, VectType& z_inout )  At least one of the inputs"
+                      + "  object x size " + std::to_string(check_size_x) + "  object y size "
+                      + std::to_string(check_size_y) + "  object z size " + std::to_string(check_size_z)
+                      + " is not sufficient to fit the nvertices_ values( " + std::to_string(nvertices_) + " )");
+    }
+
+    long i = 0;
+    if (dimw_ == 3) {
+        for (const auto& vit : vertices(gridView_, Dune::Partitions::all)) {
+            auto xyz_local = vit.geometry().corner(0); // vertices only have one corner
+            x_inout.data()[i] = static_cast<VT>(xyz_local[0]);
+            y_inout.data()[i] = static_cast<VT>(xyz_local[1]);
+            z_inout.data()[i] = static_cast<VT>(xyz_local[2]);
+            i++;
+        }
+    } else if (dimw_ == 2) {
+        double td = 0.0;
+        for (const auto& vit : vertices(gridView_, Dune::Partitions::all)) {
+            auto xyz_local = vit.geometry().corner(0); // vertices only have one corner
+            x_inout.data()[i] = static_cast<VT>(xyz_local[0]);
+            y_inout.data()[i] = static_cast<VT>(xyz_local[1]);
+            z_inout.data()[i] = static_cast<VT>(td);
+            i++;
+        }
+    }
+    assert(i == nvertices_); // As we are templated on the
+                             // Dune::PartitionSet<partitions>, this cannot change
+    return i;
+}
+
+template <class GridView, unsigned int partitions>
+template <typename T>
+long SimMeshDataAccessor<GridView,partitions>::
+writeGridPoints_AOS(T* xyz_inout, long max_size) const
+{
+    if (max_size < nvertices_ * 3) {
+        assert(max_size >= nvertices_ * 3);
```

Comment:
Tiny nit: We should `#include <cassert>` to bring in a declaration of the `assert` macro.